tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteTestIntegration.cs

```diff
@@ -69,3 +61,56 @@ internal static CallTargetState OnMethodBegin<TTarget, TTestMethod>(TTarget inst
         return CallTargetState.GetDefault();
     }
 }
+
+/// <summary>
+/// Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.TestMethodRunner.ExecuteTest calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter",
+    TypeName = "Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.TestMethodRunner",
+    MethodName = "ExecuteTestAsync",
+    ReturnTypeName = "System.Threading.Tasks.Task`1[Microsoft.VisualStudio.TestTools.UnitTesting.TestResult[]]",
+    ParameterTypeNames = ["Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.TestMethodInfo"],
+    MinimumVersion = "14.0.0",
+    MaximumVersion = "14.*.*",
+    IntegrationName = MsTestIntegration.IntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+#pragma warning disable SA1402
+public static class TestMethodRunnerExecuteTestIntegrationV3_9
+#pragma warning restore SA1402
+{
+    private static SkipTestMethodExecutorAsync? _itrSkipTestMethodExecutor;
+    private static SkipTestMethodExecutorAsync? _disabledSkipTestMethodExecutor;
+
+    internal static CallTargetState OnMethodBegin<TTarget, TTestMethod>(TTarget instance, TTestMethod testMethod)
+        where TTarget : ITestMethodRunnerV3_9
+        where TTestMethod : ITestMethod
+    {
+        // In order to skip a test we change the Executor to one that returns a valid outcome without calling
+        // the MethodInfo of the test
+        if (MsTestIntegration.IsEnabled &&
+            instance.TestMethodInfo is { Executor: { } executor } testMethodInfo)
+        {
+            SkipTestMethodExecutorAsync? newExecutor = null;
+
+            if (MsTestIntegration.ShouldSkip(testMethod, out _, out _))
+            {
+                _itrSkipTestMethodExecutor ??= new SkipTestMethodExecutorAsync(executor.GetType().Assembly, IntelligentTestRunnerTags.SkippedByReason);
+                newExecutor = _itrSkipTestMethodExecutor;
+            }
+            else if (MsTestIntegration.GetTestProperties(testMethod) is { Disabled: true, AttemptToFix: false })
+            {
+                _disabledSkipTestMethodExecutor ??= new SkipTestMethodExecutorAsync(executor.GetType().Assembly, "Flaky test is disabled by Datadog.");
+                newExecutor = _disabledSkipTestMethodExecutor;
+            }
+
+            if (newExecutor is not null)
+            {
+                testMethodInfo.Executor = DuckType.CreateReverse(executor.GetType(), newExecutor);
+            }
```

Comment:
This is duplicated from above right? Is it worth extracting to a static function and calling it in both places?

Reply:
is not the same, there's some changes over the SkipTestMethodExecutor... `instance.TestMethodInfo`  is also different. I need to think how to refactor this in away that reduces code and not end up with a bigger file :D

Reply:
Haha, ok, my bad, was only because it looked like it at first glance!! 😄